lib/ansible/modules/cloud/opennebula/one_host.py

```diff
@@ -0,0 +1,267 @@
+#!/usr/bin/python
+#
+# Copyright 2018 www.privaz.io Valletech AB
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: one_host
+
+short_description: Manages OpenNebula Hosts
+
+version_added: "2.6"
+
+requirements:
+    - pyone
+
+description:
+    - "Manages OpenNebula Hosts"
+
+options:
+    name:
+        description:
+            - Hostname of the machine to manage.
+        required: true
+    state:
+        description:
+            - Takes the host to the desired lifecycle state.
+            - If C(absent) the host will be deleted from the cluster.
+            - If C(present) the host will be created in the cluster (includes C(enabled), C(disabled) and C(offline) states).
+            - If C(enabled) the host is fully operational.
+            - C(disabled), e.g. to perform maintenance operations.
+            - C(offline), host is totally offline.
+        choices:
+            - absent
+            - present
+            - enabled
+            - disabled
+            - offline
+        default: present
+    im_mad_name:
+        description:
+            - The name of the information manager, this values are taken from the oned.conf with the tag name IM_MAD (name)
+        default: kvm
+    vmm_mad_name:
+        description:
+            - The name of the virtual machine manager mad name, this values are taken from the oned.conf with the tag name VM_MAD (name)
+        default: kvm
+    cluster_id:
+        description:
+            - The cluster ID.
+        default: 0
+    cluster_name:
+        description:
+            - The cluster specified by name.
+    template:
```

Comment:
Since we have this `template` param to be able to change labels/attributes,it would probably be nice to later add `labels` and `attributes` parameters to make it easier on the user.

Reply:
labels is planned. as per design guidelines. 
what is attributes? isnt it the template but in K=V format?

Reply:
Yeah, attributes are that. That's how they are displayed in Sunstone  and that's how we called them for one_vm and one_service modules.

It's cool that those two are planned, they don't have to be here from day 1.

Reply:
Attributes are KV fields in USER_TEMPLATE section.
Labels are special attribute stored as `LABELS` key and a list of comma separated strings which represents labels.
```
TEMPLATE = [
    LABELS = "label1,label2",
    KEY1 = VALUE1,
    KEY2 = VALUE2
]
```

Reply:
If I remember correctly in our modules `attributes` only support `key=value` (check the docs and examples: https://docs.ansible.com/ansible/devel/modules/one_vm_module.html), we don't support any advance structures, compared to your template, because we want the `attributes` to be used just for that. But yeah they are similar.